app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt

```diff
@@ -117,26 +125,42 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback,  GoogleMap.OnMarke
             if (location != null) {
                 lastLocation = location
                 val currentLatLng = LatLng(location.latitude, location.longitude)
-                map.animateCamera(CameraUpdateFactory.newLatLngZoom(currentLatLng, 12f))
+                map.moveCamera(CameraUpdateFactory.newLatLngZoom(currentLatLng, 12f))
             }
         }
 
         drawBuildingPolygons()
         map.setOnPolygonClickListener(this)
+
+
+        map.setOnMapClickListener {
+
+            //Dismiss the bottom sheet when clicking anywhere on the map
+            if (bottomSheetBehavior.state == BottomSheetBehavior.STATE_EXPANDED)
+                bottomSheetBehavior.state = BottomSheetBehavior.STATE_COLLAPSED
+        }
+
     }
 
     //implements methods of interface GoogleMap.GoogleMap.OnPolygonClickListener
     override fun onPolygonClick(p: Polygon)
     {
-        //Prints the name of the building
-        Toast.makeText(this, p.tag.toString(), Toast.LENGTH_LONG).show()
 
+        //Expand the bottom sheet when clicking on a polygon
+        //TODO: Limt only to campus buildings as poylgons could highlight anything
+        if (bottomSheetBehavior.state != BottomSheetBehavior.STATE_EXPANDED)
+            bottomSheetBehavior.state = BottomSheetBehavior.STATE_EXPANDED
+
+        //Populate the bottom sheet with building information
+
```

Comment:
Remove this spacing so that the comment can be easily associated with the code that it relates to, because in this situation it's hard to discern whether this code is related to the code that it already written, or to something not yet written

Reply:
Will do